test/Microsoft.Azure.Devices.Edge.Test.Common/IPlatform.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Test.Common
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Security.Cryptography.X509Certificates;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.Test.Common.Certs;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public interface IPlatform
+    {
+        Task<string> CollectDaemonLogsAsync(DateTime testStartTime, string filePrefix, CancellationToken token);
+
+        IEdgeDaemon CreateEdgeDaemon(Option<string> installerPath);
+
+        // After calling this function, the following files will be available under {scriptPath}:
+        //  certs/iot-edge-device-{deviceId}-full-chain.cert.pem
+        //  private/iot-edge-device-{deviceId}.key.pem
+        Task<EdgeCertificates> GenerateEdgeCertificatesAsync(string deviceId, string scriptPath, CancellationToken token);
```

Comment:
I am thinking EdgeDaemon should have a property of IPlatform; and has GenerateCertificatesAsync method; IPlatform should only have a method for generate certificate in a specific platform, it should not care about what kind of certificate.  So GenerateLeafCertificatesAsync should move to LeafDevice to generate certificate for LeafDevice.

Reply:
I don't understand what you mean. The thing that generates a certificate needs to know two things: the platform (because we use different scripts, depending on the platform) and the type of certificate (because we pass different commands to the scripts, based on certificate type).

And the daemon is not responsible for generating certificates; the certificate authority is.